src/fiber/fiber.ml

```diff
@@ -360,6 +368,55 @@ let parallel_iter_set (type a s)
   | 1 -> f (Option.value_exn (S.min_elt t)) k
   | n -> parallel_iter_generic ~n ~iter:(S.iter t) ~f k
 
+module Make_map_traversals (Map : Map.S) = struct
+  let parallel_iter t ~f k =
+    match Map.cardinal t with
+    | 0 -> k ()
+    | 1 -> Map.iteri t ~f:(fun x y -> EC.apply2 f x y k)
+    | n ->
+      EC.add_refs (n - 1);
+      let left_over = ref n in
```

Comment:
Can `parallel_iter_generic` be used here?

Reply:
We can, but I feel like with the currification/uncurrification it doesn't make the code clearer. I don't feel strongly about it though, so happy to change it if you think it's worth doing.

While looking at this code I noticed that the `1` case of `parallel_iter` and `parellel_map` where more complicated than they should and simplified them. So thanks for commenting there in any case.